src/redux/modules/spinner/spinner.js

```diff
@@ -0,0 +1,63 @@
+/* @flow */
+import debug from 'debug';
+
+type SpinnerState = {
+  canShow: boolean;
+  messageId: ?string;
+};
+
+type SpinnerAction = {
+  type: Symbol;
+  state: SpinnerState;
+};
+
+export const SHOW_SPINNER: Symbol = Symbol('@@spinner/SHOW_SPINNER');
+export const HIDE_SPINNER: Symbol = Symbol('@@spinner/HIDE_SPINNER');
+
+export const DEFAULT_SPINNER_STATE: SpinnerState = {
+  canShow: false,
+  messageId: null,
+};
+
+if (__DEBUG__) {
+  debug.enable('spinner-reducer:*');
+}
+
+const log = debug('spinner-reducer:debug');
+
+export function showSpinner(messageId: string): SpinnerAction {
+  return {
+    type: SHOW_SPINNER,
+    state: {
+      canShow: true,
+      messageId,
+    },
+  };
+}
+
+export function hideSpinner(): SpinnerAction {
+  return {
+    type: HIDE_SPINNER,
+    state: DEFAULT_SPINNER_STATE,
+  };
+}
+
+export function spinnerReducer(state : SpinnerState = DEFAULT_SPINNER_STATE, action: any): any {
```

Comment:
Reducers always return state, so `SpinnerState` in this case since you already have it.